hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -290,23 +280,24 @@ private void buildApiNodeToIndexMap(List<ApiNode<Event>> apiNodes) {
     return Optional.empty();
   }
 
-  private Table<Integer, Integer, Edge> buildTraceEdgeTable() {
-    Table<Integer, Integer, Edge> traceEdgeTable = HashBasedTable.create();
+  private void buildApiNodeToIndexMap() {
+    for (int i = 0; i < apiNodeList.size(); i++) {
```

Comment:
can we convert this for() to apiNodeList.foreach ?

Reply:
I will do it here https://github.com/hypertrace/hypertrace-ingester/pull/144

Reply:
Actually we need the index here.